src/dbnode/client/session.go

```diff
@@ -2400,7 +2420,18 @@ func (s *session) streamBlocksFromPeers(
 			enqueueCh.trackProcessed(1)
 		}
 	)
-	for perPeerBlocksMetadata := range enqueueCh.get() {
+	enqueueChInputs, err := enqueueCh.get()
+	if err != nil {
```

Comment:
We should probably make this method return an error now and bubble that as a "real" error to the caller (i.e. cancel and immediate return).

Otherwise you could "successfully" bootstrap everything even though you weren't even able to attempt fetching anything, the caller has no idea streaming didn't even begin.

Reply:
@robskillington Yeah sure. Would this trigger a retry?

Reply:
So if error is returned from `FetchBootstrapBlocksFromPeers(...)` it propagates to the peers boostrapper as an error and the time range is marked as unfulfilled - which is logged for now and returns an error.

Just capturing what we briefly discussed, we should add a brief retry loop to `storage/bootstrap.go` (unless first bootstrap, in which case it should not update it's state to `Bootstrapped` and return the error as normal) which can be determined by looking at the `d.bootstraps` counter in the `storage/database.go`.